qemu/tests/qemu_guest_agent.py

```diff
@@ -1729,16 +1736,30 @@ def _change_bl(self, session):
         Now only linux guest has this behavior,but still leave interface
         for windows guest.
         """
+
         if self.params.get("os_type") == "linux":
-            cmd_black_list = self.params["black_list"]
-            cmd_blacklist_backup = self.params["black_list_backup"]
+            cmd_blacklist_backup = self.params["black_file_backup"]
             session.cmd(cmd_blacklist_backup)
+            full_qga_ver = self._get_qga_version(session, self.vm, main_ver=False)
+            full_qga_ver = version.parse(full_qga_ver)
+            value_full_qga_ver = (full_qga_ver >= version.parse("8.1.0-5"))
+            black_list_spec = "BLACKLIST_RPC"
+            if full_qga_ver >= version.parse("8.1.0-5"):
+                black_list_spec = "allow-rpcs"
+            elif full_qga_ver >= version.parse("7.2.0-4"):
+                black_list_spec = "BLOCK_RPCS"
+            if black_list_spec == "allow-rpcs":
+                black_list_change_cmd = "sed -i 's/allow-rpcs.*/allow-rpcs=%s\"/g' /etc/sysconfig/qemu-ga"
+            else:
+                black_list_change_cmd = "sed -i 's/%s//g' /etc/sysconfig/qemu-ga"
+            cmd_black_list = self.params.get("black_list_new" if value_full_qga_ver else "black_list")
             for black_cmd in cmd_black_list.split():
-                bl_check_cmd = self.params["black_list_check_cmd"] % black_cmd
-                bl_change_cmd = self.params["black_list_change_cmd"] % black_cmd
+                bl_check_cmd = "grep -nr '%s=%s' /etc/sysconfig/qemu-ga" % (black_list_spec, black_cmd)
```

Comment:
Hi @6-dehan 

Would you please share more why we do not use the value in the cfg?Thanks.

Reply:
cleaned it.